src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Error.cs

```diff
@@ -93,13 +92,13 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
                 return false;
             }
 
-            var argumentKindType = names.First(tn => tn.Name == requiredKindField.Name).Type;
+            var argumentKindType = names.FirstOrDefault(tn => tn.Name == requiredKindField.Name).Type;
             if (argumentKindType.IsEnum)
             {
                 argumentKindType = argumentKindType.GetEnumSupertype();
             }
 
-            if (argumentKindType.Kind != requiredKindField.Type.Kind)
+            if (!(requiredKindField.Type.Kind == DKind.Number) && (argumentKindType.Kind != requiredKindField.Type.Kind))
```

Comment:
>!(requiredKindField.Type.Kind == DKind.Number) [](http://example.com/codeflow?start=16&length=46)

This will always be false, so this `if` will never be executed:

- requiredKindField is defined as the 'Kind' field of acceptedFields (line 55)
- acceptedFields is defined as the fields from reifiedError (line 54)
- reifiedError is defined from ErrorType, and the schema defines Kind as a numeric field (ErrorType.cs, line 27)

Reply:
you are right, removed this. thanks